sdks/go/pkg/beam/io/textio/textio_test.go

```diff
@@ -40,3 +46,68 @@ func TestRead(t *testing.T) {
 	}
 
 }
+
+func TestRead(t *testing.T) {
+	p, s := beam.NewPipelineWithRoot()
+	lines := Read(s, testFilePath)
+	passert.Count(s, lines, "NumLines", 1)
+
+	if _, err := beam.Run(context.Background(), "direct", p); err != nil {
+		t.Fatalf("Failed to execute job: %v", err)
+	}
+}
+
+func TestReadAll(t *testing.T) {
+	p, s := beam.NewPipelineWithRoot()
+	files := beam.Create(s, testFilePath)
+	lines := ReadAll(s, files)
+	passert.Count(s, lines, "NumLines", 1)
+
+	if _, err := beam.Run(context.Background(), "direct", p); err != nil {
+		t.Fatalf("Failed to execute job: %v", err)
+	}
+}
+
+func TestWrite(t *testing.T) {
+	out := "text.txt"
+	p, s := beam.NewPipelineWithRoot()
+	lines := Read(s, testFilePath)
+	Write(s, out, lines)
+
+	if _, err := beam.Run(context.Background(), "direct", p); err != nil {
+		t.Fatalf("Failed to execute job: %v", err)
+	}
+
+	if _, err := os.Stat(out); errors.Is(err, os.ErrNotExist) {
+		t.Fatalf("Failed to write %v", out)
+	}
+	defer os.Remove(out)
```

Comment:
Prefer using [t.Cleanup](https://pkg.go.dev/testing#T.Cleanup) instead of defer in tests, since then the cleanup will happen after test completion.

In this case it's identical (when the function scope is exited), but it can occasionally not be.

Reply:
Updated!